application/objs/paginationtoken.go

```diff
@@ -0,0 +1,68 @@
+package objs
+
+import (
+	"fmt"
+
+	"github.com/MadBase/MadNet/application/objs/uint256"
+	"github.com/MadBase/MadNet/errorz"
+)
+
+type PaginationToken struct {
+	LastPaginatedType LastPaginatedType
+	TotalValue        *uint256.Uint256
+	LastKey           []byte
+}
+
+type LastPaginatedType byte
+
+const (
+	LastPaginatedUtxo LastPaginatedType = iota
+	LastPaginatedDeposit
+)
+
+// UnmarshalBinary takes a byte slice and returns the corresponding
+// PaginationToken object
+func (pt *PaginationToken) UnmarshalBinary(data []byte) error {
+	if pt == nil {
+		return errorz.ErrInvalid{}.New("not initialized")
+	}
+
+	if data == nil || len(data) < 65 || data[0] > 1 {
+		return errorz.ErrInvalid{}.New("bytes invalid")
+	}
+
+	pt.LastPaginatedType = LastPaginatedType(data[0])
+
+	TotalValue := &uint256.Uint256{}
+	TotalValue.UnmarshalBinary(data[1:33])
+	pt.TotalValue = TotalValue
+
+	pt.LastKey = make([]byte, 0, 96)
```

Comment:
This definition leads to leading zero bytes due to appending with make(<>,0,96) ... should you be appending runes?

Reply:
I don't think we need runes as we're dealing with plain bytes and not unicode points. Also I'm just setting the capacity, the length is still 0. As far as I know that doesn't lead to leading zeros. Would you still want me to change that?